src/public/modules/core/components/banner.client.component.js

```diff
@@ -26,4 +33,21 @@ function bannerController() {
       vm.bannerHidden = true
     })
   }
+
+  const processBannerMessage = () => {
+    // Retrieve banner type from message, but it is possible that no types
+    // exist.
+    const vmMessage = vm.message || ''
+    const type = vmMessage.split(':').shift().trim().toLowerCase()
+    const retrievedType = vm.BANNER_TYPES[type]
+
+    vm.bannerType = retrievedType || vm.BANNER_TYPES.info
+
+    // If there is a type retrieved from message, remove the type encoding in
+    // the message.
+    // The + 1 is to also remove the semicolon from the encoding
+    vm.bannerMessage = retrievedType
+      ? vmMessage.substring(vmMessage.indexOf(':') + 1).trim()
+      : vmMessage
```

Comment:
I'm wondering if it might be safer to not show the banner at all if the format is incorrect. e.g. let's say instead of typing "info:here's my banner message", the dev types "infp:here's my banner message". Then the typo will show up in the banner, right?

Reply:
Don't know leh, there's merits on both sides. The current way removes the need for a prefix if the banner is informational, since it defaults to `info` if no type is provided. 

Both scenarios result in the developers checking the banner text and realizing it is wrong/missing, so I think the current way allows for more flexibility

Reply:
okay fair enough